Add a filter `woocommerce_admin_terms_metabox_datalimit` to change the data-limit value for the attributes term box

Babylon1999 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #37548 .

The current attribute term list in the product editor imposes a limitation of 50 terms creating inconvenience for merchants with attributes exceeding this limit. Consequently, they are unable to search for terms from the drop-down menu.



This proposed solution allows for a flexible adjustment of the term limit by introducing a filter, `woocommerce_admin_terms_metabox_datalimit`, which will allow to modify the data-limit variable. This approach eliminates the need to edit the plugin files directly, providing a more streamlined and customizable solution.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Import this product with 60 terms: https://d.pr/f/SArtLx
2. If you check the drop-down list in All Product > Product > Attributes, it will be limited to 50
<img src="https://d.pr/i/cuByiw+" alt="">
3. To increase the limit to 60 or more, add the following in the functions.php file:

```
function example_callback() {
    return 60;
}
add_filter( 'woocommerce_admin_terms_metabox_datalimit', 'example_callback' );
```

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Message <!-- Add a changelog message here -->

Add a filter to edit the data-limit value of the term list in the product edit page

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

Babylon1999 committed:
Update html-product-attribute-inner.php

Add a new filter `woocommerce_admin_terms_metabox_datalimit` to change the value of data-limit. See #37548 for more context.

nathanss commented:
@Babylon1999 Please also check the [linter results](https://github.com/woocommerce/woocommerce/actions/runs/6855612823/job/18641059361?pr=41411), there are some issues. The new hook should accompany some documentation. You can find a lot of examples throughout the codebase. Please add `@since 8.5.0`, which will be the next WooCommerce version.

Babylon1999 commented:
Thanks for the feedback! I added the comment, @nathanss.

As for the channel log file, shouldn't this take care of it? :\ 

<img width="1156" alt="image" src="https://github.com/woocommerce/woocommerce/assets/67080558/125e0248-a86b-4df7-9b6b-aee284dc62d7">

lsinger commented on plugins/woocommerce/includes/admin/meta-boxes/views/html-product-attribute-inner.php:
```diff
@@ -37,10 +37,17 @@
+				   * @since 8.5.0
+				   * @param int $term_limit The maximum number of terms to display in the list.
+				   */
+					$term_limit = apply_filters( 'woocommerce_admin_terms_metabox_datalimit', 50 );
```
Could you wrap the call to `apply_filters` in `absint`? Like so: 


```suggestion
					$term_limit = absint( apply_filters( 'woocommerce_admin_terms_metabox_datalimit', 50 ) );
```

lsinger commented on plugins/woocommerce/includes/admin/meta-boxes/views/html-product-attribute-inner.php:
```diff
@@ -37,10 +37,17 @@
+				  /**
+				   * Filter the length (number of terms) rendered in the list.
+	               *
+				   * @since 8.5.0
```
Please update to `8.6.0`: 

```suggestion
				   * @since 8.6.0
```

lsinger reviewed (changes requested):
Thank you @Babylon1999, looks good! I left a few comments to address, and the linter is [also still complaining](https://github.com/woocommerce/woocommerce/actions/runs/6897952450/job/18767112098?pr=41411), as [previously pointed out](https://github.com/woocommerce/woocommerce/pull/41411#issuecomment-1810195393). Please ensure to get the linter to pass:

```
FILE: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-attribute-inner.php
-----------------------------------------------------------------------------------------------
FOUND 2 ERRORS AND 6 WARNINGS AFFECTING 8 LINES
-----------------------------------------------------------------------------------------------
 40 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
 40 | ERROR | Line indented incorrectly; expected at least 5 tabs, found 4 tabs and 2 spaces (Generic.WhiteSpace.ScopeIndent.Incorrect)
 41 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
 42 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
 42 | ERROR | Tabs must be used to indent lines; spaces are not allowed (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 43 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
 44 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
 45 | WARNING | Found precision alignment of 2 spaces. (WordPress.WhiteSpace.PrecisionAlignment.Found)
-----------------------------------------------------------------------------------------------
```

To run the linter locally, commit your changes and then run `pnpm lint:php:changes:branch`. Please let me know if I can assist in any other way.

Babylon1999 commented:
Hello @lsinger,

I finished the requested changes but the linter actions are still nagging. I replaced the spaces with tabs and everything, I feel lost at this point. 

Also, I saw an error about a missing channel log file when I already added one.

Babylon1999 commented:
I'm sorry for the mess I created, I didn't think adding two lines required all this testing. I have maintainers edit enabled, so you can edit the changes or let me know what I'm doing wrong. (._.`)

M0nkfiish commented:
Just following up on this to see if any progress? Looks like it's close to being ready? - I have a clients site that I have to keep editing the core WP file to make this work but needs doing every few weeks due to WC updates so getting pretty annoying :)

Babylon1999 committed:
Update html-product-attribute-inner.php

Babylon1999 committed:
Creating a changelog file again

lsinger commented:
Moved to #45506. ✅
